skimage/filters/thresholding.py

```diff
@@ -271,16 +271,15 @@ def threshold_otsu(image, nbins=256):
     -----
     The input image must be grayscale.
     """
-    if len(image.shape) > 2 and image.shape[-1] in (3, 4):
+    if image.ndim > 2 and image.shape[-1] in (3, 4):
         msg = "threshold_otsu is expected to work correctly only for " \
               "grayscale images; image shape {0} looks like an RGB image"
         warn(msg.format(image.shape))
 
     # Check if the image is multi-colored or not
-    if image.min() == image.max():
-        raise ValueError("threshold_otsu is expected to work with images "
-                         "having more than one color. The input image seems "
-                         "to have just one color {0}.".format(image.min()))
+    first_pixel = image.ravel()[0]
```

Comment:
I feel like I'm the grumpy conservative person here but this change does not bring an improvement on my machine
```python
In [20]: arr = np.zeros((10000, 10000))   

In [21]: %timeit arr.min() == arr.max()                                                                                             
54.2 ms ± 168 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [22]: %timeit np.all(arr == arr.ravel()[0])                                                                                      
60.5 ms ± 334 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```
Is it possible to see more benchmarks?

Reply:
On my machine:
```python
In [1]: arr = arr = np.zeros((10000, 10000))                                                             

In [2]: %timeit arr.min() == arr.max()                                                                   
57.7 ms ± 119 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [3]: %timeit np.all(arr == arr.ravel()[0])                                                            
28.9 ms ± 19.8 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

Reply:
I've also rerun on my Linux machine, and done it backwards to ensure there's nothing clock speed related going on

```
arr = np.zeros((10000, 10000))

%timeit np.all(arr == arr.ravel()[0])
32.8 ms ± 1.42 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit arr.min() == arr.max()
43.5 ms ± 1.25 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

Reply:
I also see a speedup, as noted on the issue... @emmanuelle your computer is the weird one here it seems! 😬 But yes, it would be good to get to the bottom of this discrepancy. Logically, I have agree that one pass should be cheaper than two passes!